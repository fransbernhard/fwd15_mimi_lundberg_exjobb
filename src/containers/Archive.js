import React, { PureComponent } from 'react';

import Menu from '../components/Menu';
import Footer from '../components/Footer';
import ProductContainer from '../components/ProductContainer';
import CategoryContainer from '../components/CategoryContainer';

class Archive extends PureComponent {
    state = {
        products: [],
        category: "",
        activeCategoryIndex: 3
    }

    filterHandler = (category, index) => {
        this.setState({
            category,
            activeCategoryIndex: index
        })
    }

    componentDidMount(){
        const myInit = {
            method: "GET",
            headers: {
                "Content-Type": "application/json",
                'Accept': 'application/json'
            }
        };

        // fetch("./php/products.php", myInit)
        fetch("/getProducts", myInit)
            .then((res) => res.json())
            .then((data) => {
                this.setState({ products: data });
            }).catch(function(err) {
                console.log('Error fetching products: ' + err.message);
            });
    }

    render() {
        const { category, activeCategoryIndex, products } = this.state;

        return (
            <div>
                <Menu/>
                <div className="archive-container" id="archive">
                    <div className="archive-wrapper">
                        <CategoryContainer
                            filterHandler={this.filterHandler}
                            products={products}
                            activeCategoryIndex={activeCategoryIndex}
                        />
                        <br/><br/>
                        <ProductContainer
                            products={category.length
                                ? products.filter((prod) => prod.catName === category)
                                : products.filter((prod) => prod.catName === 'Places')
                            }
                        />
                    </div>
                </div>
                <Footer/>
            </div>
        )
    }
}

export default Archive;
