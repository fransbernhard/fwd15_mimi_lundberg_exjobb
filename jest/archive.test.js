import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import Archive from '../src/archive';

describe('components', () => {
  describe('<Archive/>', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <Archive />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
}); 
