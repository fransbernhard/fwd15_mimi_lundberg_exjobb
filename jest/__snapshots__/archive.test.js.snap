// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <Archive/> renders correctly 1`] = `
<div>
  <header>
    <nav>
      <ul>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            About
          </a>
        </li>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            Archive
          </a>
        </li>
        <li
          className="contactMobile"
        >
          <a
            href="mailto:magdamargaretha@gmail.com?Subject=magdenmagden"
          >
            Contact
          </a>
        </li>
        <li
          className="contactWeb"
        >
          <a
            onClick={[Function]}
            style={Object {}}
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    className="archive-container"
  >
    <div
      className="archive-wrapper"
    >
      <div>
        <ul
          className="filterList"
        >
          <div>
            <li
              className="category"
              onClick={[Function]}
            >
              paint
            </li>
          </div>
          <div>
            <li
              className="category"
              onClick={[Function]}
            >
              print
            </li>
          </div>
          <div>
            <li
              className="category"
              onClick={[Function]}
            >
              places
            </li>
          </div>
        </ul>
      </div>
      <br />
      <br />
      <div>
        <div
          className="prodContainer"
        >
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            23-sm.png
            <h3>
              clowd
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            24-sm.png
            <h3>
              dålig sikt
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            25-sm.png
            <h3>
              dålig sikt
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            1-sm.png
            <h3>
              pink
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            27-sm.png
            <h3>
              pink
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            2-sm.png
            <h3>
              pinks
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            3-sm.png
            <h3>
              pink
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            4-sm.png
            <h3>
              pinks
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            5-sm.png
            <h3>
              pinks
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            19-sm.png
            <h3>
              strechmarks
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            20-sm.png
            <h3>
              clowd
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            21-sm.png
            <h3>
              dålig sikt
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            22-sm.png
            <h3>
              dålig sikt
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            vertical-sm.jpg
            <h3>
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                vertical
              </span>
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            optics-sm.jpg
            <h3>
              optics
            </h3>
            <p>
              matt emulsion/akryl
            </p>
            <p>
              100x150
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            moby-sm.jpg
            <h3>
              moby
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            twink-sm.jpg
            <h3>
              twink
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            kaft-sm.jpg
            <h3>
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                käft
              </span>
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            tvatttid-sm.jpg
            <h3>
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                tvätt tid
              </span>
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            piotr-sm.jpg
            <h3>
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                piotr
              </span>
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            storabla-sm.jpg
            <h3>
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                stora blå
              </span>
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            strechmarks-sm.jpg
            <h3>
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                strechmarks
              </span>
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            clowd-sm.jpg
            <h3>
              clowd
            </h3>
            <p>
              matt emulsion
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
          <div
            className="product hvr-sink"
            onClick={[Function]}
          >
            <noscript />
            daligsikt-sm.jpg
            <h3>
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                dålig sikt
              </span>
            </h3>
            <p>
              matt emulsion/olja/akryl
            </p>
            <p>
              100x130
               cm
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer>
    <a
      href="mailto:magdamargaretha@gmail.com"
    >
      magdamargaretha@gmail.com, 
    </a>
    <a
      href="tel:+46735139012"
    >
      +46735139012
    </a>
    <p>
      Kristianstadsgatan 7, 334 56 Malmö
    </p>
  </footer>
</div>
`;
